### PR TITLE
jwt-logout was implemented

### PR DESCRIPTION
**Pull request by olyaklipa**
CustomLogoutHandler that was commented out to discuss...

### CODE REVIEW

File: src/main/java/olya/app/remindme/controller/AuthController.java [hunk 1]
```diff
@@ -5,16 +5,15 @@
...
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequiredArgsConstructor
-@RequestMapping("/auth")
+@RequestMapping("api/auth")
```
Comment (pauldaniv) [inline src/main/java/olya/app/remindme/controller/AuthController.java]: Make endpoint declarations start with `/`

**Review by pauldaniv — CHANGES_REQUESTED** [review]

File: src/main/java/olya/app/remindme/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -19,10 +21,12 @@ public class JwtAuthenticationFilter extends OncePerRequestFilter {
     private static final String BEARER = "Bearer ";
     private final JwtDecoder jwtDecoder;
     private final JwtToPrincipalConverter jwtToPrincipalConverter;
+    private final TokenRepository tokenRepository;
 
     @Override
     protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
         extractTokenFromRequest(request)
+                .filter(jwt -> !isTokenRevoked(jwt))
```
Comment (pauldaniv) [inline src/main/java/olya/app/remindme/security/JwtAuthenticationFilter.java]: Move this down after token validation

File: src/main/java/olya/app/remindme/service/impl/AuthServiceImpl.java [hunk 3]
```diff
@@ -1,38 +1,75 @@
...
+
+    private void revokeUserTokens(User user) {
+        List<Token> userTokens = tokenRepository.findByUserIdAndRevokedFalse(user.getId());
+        if (userTokens.isEmpty())
+            return;
+        userTokens.forEach(t -> t.setRevoked(true));
+        tokenRepository.saveAll(userTokens);
+    }
+
+    @Scheduled(cron = "0 34 19 * * SUN")
```
Comment (pauldaniv) [inline src/main/java/olya/app/remindme/service/impl/AuthServiceImpl.java]: Why does this ren only on Sunday?

**Review by pauldaniv — CHANGES_REQUESTED** [review]

File: src/main/java/olya/app/remindme/service/impl/AuthServiceImpl.java (hunk 3 above)
Comment (olyaklipa) [reply src/main/java/olya/app/remindme/service/impl/AuthServiceImpl.java]: I was testing it on Sunday. But there is nothing particular with a week day. Can be any or everyday (*) if we want.

**Review by pauldaniv — APPROVED** [review]

**Pull request merged**